src/menu/MenuItem.ts

```diff
@@ -0,0 +1,176 @@
+import { assign } from '@dojo/core/lang';
+import { v } from '@dojo/widget-core/d';
+import { VirtualDomProperties } from '@dojo/widget-core/interfaces';
+import ThemeableMixin, { theme, ThemeableProperties } from '@dojo/widget-core/mixins/Themeable';
+import WidgetBase from '@dojo/widget-core/WidgetBase';
+import * as css from './styles/menu.m.css';
+
+export type MenuItemType = 'item' | 'checkbox' | 'radio';
+
+/**
+ * @type MenuItemProperties
+ *
+ * Properties that can be set on a MenuItem component.
+ *
+ * @property active			Determines whether the item should receive focus.
+ * @property controls		The ID of an element that this input controls.
+ * @property disabled		Indicates whether the item is disabled.
+ * @property expanded		Indicates whether a widget controlled by `this` is expanded.
+ * @property focusable		Determines whether the item can receive focus with tab key.
+ * @property hasMenu		Indicates whether the widget is used as the label for a menu.
+ * @property hasPopup		Indicates whether the widget has a drop down child.
+ * @property id				Specifies the ID for the widget.
+ * @property index			Specifies the index of the item within a parent menu.
+ * @property menuId			The ID of the parent menu.
+ * @property onClick		Called when the widget is activated via a click or space key.
+ * @property onKeyDown		Called when keys are pressed while the widget has focus.
+ * @property properties		Additional properties for the widget's vnode.
+ * @property selected		Determines whether the widget is marked as selected.
+ * @property tag			The HTML tag name to use for the widget's vnode. Defaults to 'a'.
+ * @property type			The type of the menu item. Defaults to 'item'.
+ */
+export interface MenuItemProperties extends ThemeableProperties {
+	active?: boolean;
+	controls?: string;
+	disabled?: boolean;
+	expanded?: boolean;
+	focusable?: boolean;
+	hasMenu?: boolean;
+	hasPopup?: boolean;
+	id?: string;
+	index?: number;
+	menuId?: string;
+	onClick?: (event: Event) => void;
+	onKeyDown?: (event: KeyboardEvent) => void;
+	properties?: VirtualDomProperties;
+	selected?: boolean;
+	tag?: string;
+	type?: MenuItemType;
+}
+
+export const MenuItemBase = ThemeableMixin(WidgetBase);
+
+const SPACE_KEY = 32;
+const ENTER_KEY = 13;
```

Comment:
Agreed. I'll add a `Keys` map and the vw observer to a `common/util.ts` module, and we can expand on that/separate functionality later if the need arises.